cmd/noms/noms_sync.go

```diff
@@ -77,10 +77,21 @@ func runSync(args []string) int {
 		lastProgressCh <- last
 	}()
 
+	sourceRef := types.NewRef(sourceObj)
+	sinkRef, _ := sinkDataset.MaybeHeadRef()
+	nonFF := false
 	err = d.Try(func() {
 		defer profile.MaybeStartProfile().Stop()
+		sinkDataset.Pull(sourceStore, sourceRef, p, progressCh)
+
 		var err error
-		sinkDataset, err = sinkDataset.Pull(sourceStore, types.NewRef(sourceObj), p, progressCh)
+		if sourceRef.Height() > sinkRef.Height() {
```

Comment:
it's kind of a bummer that the only way to test this case is by testing noms_sync. What about moving this logic back into dataset.Pull() or if you think we should keep something like Pull to just move objects, then adding api to Dataset?